headers-core/src/lib.rs

```diff
@@ -23,13 +23,13 @@ pub trait Header {
     /// The name of this header.
     fn name() -> &'static HeaderName;
 
-    /// Decode this type from an iterator of `HeaderValue`s.
+    /// Decode this type from an iterator of [`HeaderValue`]s.
     fn decode<'i, I>(values: &mut I) -> Result<Self, Error>
     where
         Self: Sized,
         I: Iterator<Item = &'i HeaderValue>;
 
-    /// Encode this type to a `HeaderMap`.
+    /// Encode this type to a [`HeaderValue`].
```

Comment:
Ah, well, sorta. The argument is a container of `HeaderValue`s. Normally, it will be an private type that's like an `Entry`. But since it's generic, unit tests could just pass a `Vec<_>`.

Reply:
Added a detailed document, which would be more helpful.